src/provider/provider_level_zero.c

```diff
@@ -211,6 +211,49 @@ static umf_result_t ze2umf_result(ze_result_t result) {
     }
 }
 
+static umf_result_t ze_init_drivers(void *lib_handle, const char *lib_name) {
+    ze_result_t (*zeInitDriversFunc)(uint32_t *, ze_driver_handle_t *,
+                                     ze_init_driver_type_desc_t *);
+    *(void **)&zeInitDriversFunc =
+        utils_get_symbol_addr(lib_handle, "zeInitDrivers", lib_name);
+    if (!zeInitDriversFunc) {
+        return UMF_RESULT_ERROR_DEPENDENCY_UNAVAILABLE;
+    }
+
+    ze_init_driver_type_desc_t desc = {
+        .stype = ZE_STRUCTURE_TYPE_INIT_DRIVER_TYPE_DESC,
+        .pNext = NULL,
+        .flags = ZE_INIT_DRIVER_TYPE_FLAG_GPU};
```

Comment:
Why only GPUs?
That is why I was against calling `zeInit` or `zeInitDrivers` from UMF's code. It is the client's responsibility to initiate L0 properly.

Reply:
If we don't call zeInit/zeInitDrivers, the entire provider won't work. As for what specifically we are initializing - GPU is currently the only thing that makes sense from the UMF perspective. If there will be some other device types in the future that we want to support we can extend params to allow user to specify that.

Reply:
I do not know how user can create level zero provider without zeInit on their side.  They need to provoide context/device handle, which need level zero inited?

Reply:
+1 for what @lplewa wrote.

@igchor are there any issues when adapter statically linked with libze_loader?